thunder/torch/__init__.py

```diff
@@ -1654,6 +1654,26 @@ def convert(x, if_none):
     return result
 
 
+@torchsymbol(torch.all, is_method=True, id="torch.all")
+def all_tensor(
```

Comment:
What do you think of PyTorch's reference implementation that uses `any` to implement `all`?

https://github.com/pytorch/pytorch/blob/82b7b59d2acd849cb16f2167fa3eaf7cabf95dcf/torch/_refs/__init__.py#L2212

Reply:
Oh, I did check that implementation out but thought it is more urgent to implement `torch.all` first. Do you think it is a better idea to submit a separate PR for `torch.any`?

Reply:
If you think the implementation of `all` in terms of `any` makes sense, then this PR could include implementations for both of them since they're so similar

Reply:
Yeah, it does make sense to me. Also, considering future use cases of `torch.any`, I think it'll be better to implement `torch.any` and use it for `torch.all`. Let me add this to this PR. Thank you for the suggestion!